marconi-chain-index/test/Spec/Marconi/ChainIndex/Experimental/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -0,0 +1,556 @@
+
+-- | The property we test here is that:
+--   * mixedIndexer and listIndexer will provide the same queryResults for the same stored events
+--   * mixedIndexer's components, in-memory and SQLite, combined will provide the same queryResult as the ListIndexer
+propMixedIndexerAndListIndexerProvideTheSameQueryResult :: Property
+propMixedIndexerAndListIndexerProvideTheSameQueryResult = property $ do
+  cp <- forAll $ genChainPoint' genBlockNo genSlotNo -- generate some non genesis chainpoints
+  timedUtxoEvent :: Core.TimedEvent Utxo.UtxoEvent <-
+    forAll $ genShelleyEraUtxoEventsAtChainPoint cp
+  conn <- liftIO $ Utxo.initSQLite  ":memory:"
+  let
+    (keep,flush) = (0,0) -- small memory to force SQL flush
+    indexer :: Core.MixedIndexer Core.SQLiteIndexer Core.ListIndexer Utxo.UtxoEvent
+    indexer = Utxo.mkMixedIndexer' conn keep flush
+  mixedIndexer <- Core.index' timedUtxoEvent indexer
+    >>= Hedgehog.evalEither
+
+  listIndexer :: Core.ListIndexer Utxo.UtxoEvent <-
+    Core.index timedUtxoEvent Core.listIndexer -- add events to in-memory listIndexer
+  [utxosSQLCount] <- liftIO
+    (SQL.query_ conn "SELECT count(1) from unspent_transactions" :: IO [Integer])
+  let
+    unprocessedUtxos :: [Core.TimedEvent Utxo.Utxo]
+    unprocessedUtxos  = Utxo.timedUtxosFromTimedUtxoEvent timedUtxoEvent
+
+    queryUtxos :: [Utxo.QueryUtxoByAddress]
+    queryUtxos = mkQuery unprocessedUtxos cp
+        -- test the mixedIndexer
+  forM_ queryUtxos $ \q ->
+    do
+        fromSqliteIndexer <- Core.query cp q (mixedIndexer ^. Core.inDatabase)
+        fromMixedIndexer <- Core.query cp q mixedIndexer
+        fromMixedIndexerMem <- Core.query cp q (mixedIndexer ^. Core.inMemory )
+        fromListIndexer <- Core.resumeResult cp q listIndexer (pure [])
+        Hedgehog.footnote
+          $  "\n== Counts =="
+          <> "\nListIndexer count: " <> show ( length $ listIndexer ^. Core.events)
+          <> "\nmixedIndexer inMemory count: " <> show (length $ mixedIndexer ^. Core.inMemory . Core.events)
+          <> "\nSQLite count unspent_transactions: " <> show utxosSQLCount
+          <> "\nMiexedIndexer: " <> show (Set.fromList fromMixedIndexer)
+          <> "\nlistIndexer: " <> show (Set.fromList fromListIndexer)
+          <> "\nmixedIndexer SQLite count: " <> show (Set.fromList fromSqliteIndexer)
+
+        -- Check we arrive at the same results from both indexers
+        Set.fromList fromListIndexer === Set.fromList fromMixedIndexer
+       -- Check that mixedIndexer's sql and in-memory indexers combined are the same as the listIndexer
+        Set.fromList fromListIndexer === Set.union
+          (Set.fromList fromSqliteIndexer)  (Set.fromList fromMixedIndexerMem)
+
+-- | The property we test here is that:
+--   * mixedIndexer's in-memory indexer behaves the same as the listIndexer
+propListIndexerAndMixedIndexerInMemroyIndexerProvideTheSameQueryResult :: Property
```

Comment:
@koslambrou this test should compares the mixedIndexers in-memory indexer with that of the listIndexer as we discussed in the code walk-thru